src/hooks/useReactEventPlugin.tsx

```diff
@@ -0,0 +1,51 @@
+import type { DOMEventMap } from "prosemirror-view";
+import { useCallback, useMemo, useState } from "react";
+
+import { createReactEventPlugin } from "../plugins/createReactEventPlugin.js";
+import type { EventHandler } from "../plugins/createReactEventPlugin.js";
+
+/**
+ * Produces a plugin that can be used with ProseMirror to handle DOM
+ * events at the EditorView.dom element.
```

Comment:
better as "at the editor mount point" or nah?

Reply:
I genuinely don't know haha. What do you think? I think both are going to be equally confusing, initially